icu4j/main/classes/core/src/com/ibm/icu/impl/number/MutablePatternModifier.java

```diff
@@ -400,10 +401,21 @@ public CharSequence getSymbol(int type) {
                 return currency.getCurrencyCode();
             } else if (unitWidth == UnitWidth.HIDDEN) {
                 return "";
+            } else if (unitWidth == UnitWidth.SHORT) {
+                // Use currency.getName() here to make custom symbols work
+                return currency.getName(symbols.getULocale(), Currency.SYMBOL_NAME, null);
             } else {
-                int selector = unitWidth == UnitWidth.NARROW ? Currency.NARROW_SYMBOL_NAME
-                        : Currency.SYMBOL_NAME;
-                return currency.getName(symbols.getULocale(), selector, null);
+                CurrencyDisplayNames names = CurrencyDisplayNames.getInstance(symbols.getULocale());
+                switch (unitWidth) {
+                    case NARROW:
+                        return names.getNarrowSymbol(currency.getCurrencyCode());
```

Comment:
Question: Why getName() above "to make custom symbols work" but getSpecificXyzSymbol() here?